src/main/java/training/subsystems/elbow/Elbow.java

```diff
@@ -0,0 +1,71 @@
+package training.subsystems.elbow;
+
+import com.revrobotics.CANSparkBase;
+import com.revrobotics.CANSparkLowLevel;
+import com.revrobotics.CANSparkMax;
+import edu.wpi.first.math.controller.ArmFeedforward;
+import edu.wpi.first.math.geometry.Rotation2d;
+import utils.GBSubsystem;
+
+public class Elbow extends GBSubsystem {
+
+	private final CANSparkMax motor;
+	private static Elbow instance;
+	private ArmFeedforward armFeedforward;
+
+	private Elbow() {
+		this.armFeedforward = ElbowConstants.ARM_FEEDFORWARD;
+		this.motor = new CANSparkMax(ElbowConstants.MOTOR_ID, CANSparkLowLevel.MotorType.kBrushless);
+		motor.getPIDController().setP(ElbowConstants.MOTOR_P);
+		motor.getPIDController().setI(ElbowConstants.MOTOR_I);
+		motor.getPIDController().setD(ElbowConstants.MOTOR_D);
+	}
+
+	public static void init() {
+		if (instance == null) {
+			instance = new Elbow();
+		}
+	}
+
+	public static Elbow getInstance() {
+		init();
+		return instance;
+	}
+
+	public Rotation2d getPosition() {
+		return Rotation2d.fromRotations(motor.getEncoder().getPosition());
+	}
+
+	public Rotation2d getVelocity() {
+		return Rotation2d.fromRotations(motor.getEncoder().getVelocity());
+	}
+
+
+	public void goToAngle(Rotation2d targetAngle) {
+		motor.getPIDController()
+			.setReference(
+				targetAngle.getDegrees(),
+				CANSparkBase.ControlType.kPosition,
+				ElbowConstants.PID_SLOT,
+				ElbowConstants.ARM_FEEDFORWARD.calculate(getPosition().getRadians(), getVelocity().getRotations())
+			);
+	}
+
+	public boolean isAtAngle(Rotation2d targetAngle) {
+		return Math.abs(targetAngle.getDegrees() - motor.getEncoder().getPosition()) == 0;
```

Comment:
use tolerance

Reply:
@maya1414 tolerance is the margin for error in a calculation, when running your code on the actual robot the angle will never be precisely 0, so we use tolerance to give it space for error, the tolerance should be as accurate as u want the calculation to be, for now u can put 5 or something, but when you write other code u need to think yourself what value to give it.
```suggestion
		return Math.abs(targetAngle.getDegrees() - motor.getEncoder().getPosition()) <= YourConstants.TOLERANCE;
```